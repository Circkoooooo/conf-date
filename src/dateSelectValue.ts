export type detailConfig = {
	startTime?: number
	endTime?: number
	value?: string
}[]

export type DateConfigType = {
	requireDate: string
	config: detailConfig
}

/**
 * 提供一个配置。这个requireDate YY-MM-DD HH:mm:ss
 * 2022-06-18 00:00:00
 * @param dateConfig
 * @returns
 */
export const dateSelectValue = (
	dateConfig: DateConfigType
): (string | undefined)[] => {
	const { config, requireDate } = dateConfig
	//检测date的格式是否正确。一定要满足YY-MM-DD HH:mm:ss

	//
	const data = getFullData(requireDate)
	const hour = data[3]
	// 得到了可以使用的value
	const availableValue = config.map(item => {
		if (
			item.endTime !== undefined &&
			item.startTime !== undefined &&
			item.startTime <= parseInt(hour) &&
			item.endTime >= parseInt(hour)
		) {
			return item.value
		}
	})
	return availableValue.filter(item => {
		return item !== undefined
	})
}

const getFullData = (requireDate: string): Array<string> => {
	const [date, time] = requireDate.split(' ')
	const [year, mouth, day] = date.split('-')
	const [hour, minute, second] = time.split(':')
	return [year, mouth, day, hour, minute, second]
}
