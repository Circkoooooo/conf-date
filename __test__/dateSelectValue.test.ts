import { dateSelectValue, DateConfigType } from '../src/dateSelectValue'

const dateConfig: DateConfigType = {
	requireDate: '2022-06-18 10:00:00',
	config: [
		{
			startTime: 1,
			endTime: 12,
			value: '刚刚',
		},
		{
			startTime: 2,
			endTime: 12,
			value: '不久前',
		},
	],
}

test('dateSelectValue ', () => {
	expect(dateSelectValue(dateConfig)).toEqual(['刚刚', '不久前'])
})
