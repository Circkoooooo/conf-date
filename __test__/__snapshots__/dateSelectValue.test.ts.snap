// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dateSelectValue  1`] = `Array []`;
